### PR TITLE
[IMP] mail: introduce ChannelMemberListView/channel

### PR DESCRIPTION
**Pull request by alexkuhn**
Task-2831082

### CODE REVIEW

**Review by phenix-factory — APPROVED** [review]
LGTM

File: addons/mail/static/src/components/channel_member_list/channel_member_list.js [hunk 1]
```diff
@@ -7,16 +7,16 @@ const { Component } = owl;
 export class ChannelMemberList extends Component {
 
     /**
-     * @returns {Thread}
+     * @returns {ChannelMemberListView}
      */
-    get channel() {
-        return this.messaging.models['Thread'].get(this.props.channelLocalId);
+     get channelMemberListView() {
+        return this.messaging.models['ChannelMemberListView'].get(this.props.localId);
```
Comment (phenix-factory) [inline addons/mail/static/src/components/channel_member_list/channel_member_list.js:13]: Missing `this.messaging` guard

File: addons/mail/static/src/components/channel_member_list/channel_member_list.js [hunk 2]
```diff
@@ -23,10 +16,7 @@ export class ChannelMemberList extends Component {
 }
 
 Object.assign(ChannelMemberList, {
-    props: {
-        channelLocalId: String,
-        localId: String,
-    },
+    props: {localId: String },
```
Comment (phenix-factory) [inline addons/mail/static/src/components/channel_member_list/channel_member_list.js:19]: Missing space

File: addons/mail/static/src/components/channel_member_list/channel_member_list.js (hunk 1 above)
Comment (alexkuhn) [reply addons/mail/static/src/components/channel_member_list/channel_member_list.js]: Changes done

File: addons/mail/static/src/models/channel_member_list_view.js [hunk 3]
```diff
@@ -2,11 +2,31 @@
...
+            if (this.threadViewOwner) {
+                return replace(this.threadViewOwner.thread);
+            }
+            return clear();
+        },
+    },
     fields: {
+        channel: one('Thread', {
+            compute: '_computeChannel',
+            required: true,
```
Comment (phenix-factory) [inline addons/mail/static/src/models/channel_member_list_view.js:28]: Required on compute seams strange.But maybe you want `readonly` ?

Comment (alexkuhn) [reply addons/mail/static/src/models/channel_member_list_view.js]: Changes done.

**Pull request closed without merging**